addon-manager/impl/src/main/java/org/jboss/forge/addon/manager/impl/ui/DistributionUpdateCommand.java

```diff
@@ -0,0 +1,171 @@
+import org.jboss.forge.furnace.addons.Addon;
+import org.jboss.forge.furnace.util.OperatingSystemUtils;
+import org.jboss.forge.furnace.versions.SingleVersion;
+import org.jboss.forge.furnace.versions.Version;
+
+public class DistributionUpdateCommand extends AbstractUICommand implements AddonCommandConstants
+{
+   @Inject
+   Addon currentAddon;
+   @Inject
+   private DependencyResolver dependencyResolver;
+
+   @Override
+   public UICommandMetadata getMetadata(UIContext ctx)
+   {
+      return Metadata.forCommand(getClass()).name("Forge Update Distribution")
+               .description("Update the whole forge installation")
+               .category(Categories.create(ADDON_MANAGER_CATEGORIES));
+   }
+
+   @Override
+   public Result execute(UIExecutionContext context) throws Exception
+   {
+      if (!context.getPrompt().promptBoolean(
+               "Are you sure you want to continue? This command will delete current directories: addons, bin, lib"))
+      {
+         return Results.fail("Updating distribution was aborted.");
+      }
+
+      List<Coordinate> distributionVersions = dependencyResolver.resolveVersions(DependencyQueryBuilder
+               .create(CoordinateBuilder.create()
+                        .setGroupId("org.jboss.forge")
+                        .setArtifactId("forge-distribution")));
+      Coordinate latestDistributionCoordinate;
+      int i = 0;
+      do
+      {
+         i++;
+         if (distributionVersions.size() <= i)
+         {
+            // if we haven't found final version
+            return Results.fail("Have not found any online Final version of the distribution.");
+         }
+         latestDistributionCoordinate = distributionVersions.get(distributionVersions.size() - i);
+      }
+      while (latestDistributionCoordinate.isSnapshot());
+      Version latestVersion = new SingleVersion(latestDistributionCoordinate.getVersion());
+      Version installedVersion = currentAddon.getId().getVersion();
+      if (latestVersion.compareTo(installedVersion) <= 0)
+      {
+         return Results.fail("Forge is already in the most updated version.");
+      }
+      File updateDir = new File(OperatingSystemUtils.getForgeHomeDir().getAbsolutePath() + "/.update");
+
+      Dependency dependency = dependencyResolver.resolveArtifact(DependencyQueryBuilder.create(CoordinateBuilder
+               .create()
+               .setGroupId(latestDistributionCoordinate.getGroupId())
+               .setArtifactId(latestDistributionCoordinate.getArtifactId())
+               .setVersion(latestDistributionCoordinate.getVersion()).setClassifier("offline").setPackaging("zip")));
+      FileResource<?> artifact = dependency.getArtifact();
+      ZipFile zipFile = new ZipFile(artifact.getFullyQualifiedName());
+      File tempFolder = OperatingSystemUtils.createTempDir();
+      String tempFolderPath = tempFolder.getAbsolutePath();
+      zipFile.extractAll(tempFolderPath);
```

Comment:
It is because it unzips the folder and not its content. If I would extract it into .update, there would be .update/forge-distribution-2.16.0.Final folder. Instead I will only copy into .update everything inside the forge-distribution-2.16.0.Final (playing with versions is uneccessary after this)